File: backends/php-laravel/app/Support/ValidationProblem.php
<?php

namespace App\Support;

use RuntimeException;

class ValidationProblem extends RuntimeException
{
    /**
     * @param  list<array{field: string, messageKey: string}>  $violations
     */
    public function __construct(public readonly array $violations)
    {
        parent::__construct('Validation failed');
    }
}

File: backends/spring-java/src/main/java/dev/stackverse/backend/common/ValidationProblem.java
package dev.stackverse.backend.common;

import java.util.List;

/** Validation failure carrying field-level errors (SPEC rules 5 and 11). */
public class ValidationProblem extends RuntimeException {
    private final List<FieldViolation> violations;

    public ValidationProblem(List<FieldViolation> violations) {
        super("Validation failed");
        this.violations = List.copyOf(violations);
    }

    public List<FieldViolation> getViolations() {
        return violations;
    }
}
